routers/api/v1/repo/issue.go

```diff
@@ -902,3 +903,399 @@ func UpdateIssueDeadline(ctx *context.APIContext) {
 
 	ctx.JSON(http.StatusCreated, api.IssueDeadline{Deadline: &deadline})
 }
+
+// GetIssueDependencies list an issue's dependencies
+func GetIssueDependencies(ctx *context.APIContext) {
+	// swagger:operation GET /repos/{owner}/{repo}/issues/{index}/dependencies issue issueListIssueDependencies
+	// ---
+	// summary: List an issue's dependencies
+	// produces:
+	// - application/json
+	// parameters:
+	// - name: owner
+	//   in: path
+	//   description: owner of the repo
+	//   type: string
+	//   required: true
+	// - name: repo
+	//   in: path
+	//   description: name of the repo
+	//   type: string
+	//   required: true
+	// - name: index
+	//   in: path
+	//   description: index of the issue
+	//   type: string
+	//   required: true
+	// - name: page
+	//   in: query
+	//   description: page number of results to return (1-based)
+	//   type: integer
+	// - name: limit
+	//   in: query
+	//   description: page size of results
+	//   type: integer
+	// responses:
+	//   "200":
+	//     "$ref": "#/responses/IssueList"
+
+	if !ctx.Repo.Repository.IsDependenciesEnabled() {
+		ctx.NotFound()
+		return
+	}
+
+	issue, err := models.GetIssueWithAttrsByIndex(ctx.Repo.Repository.ID, ctx.ParamsInt64(":index"))
+	if err != nil {
+		if models.IsErrIssueNotExist(err) {
+			ctx.NotFound("IsErrIssueNotExist", err)
+		} else {
+			ctx.Error(http.StatusInternalServerError, "GetIssueByIndex", err)
+		}
+		return
+	}
+
+	deps, err := issue.BlockedByDependencies()
+	if err != nil {
+		ctx.Error(http.StatusInternalServerError, "BlockedByDependencies", err)
+		return
+	}
+
+	page := ctx.FormInt("page")
+	if page <= 1 {
+		page = 1
+	}
+	limit := ctx.FormInt("limit")
+	if limit <= 1 {
+		limit = setting.API.DefaultPagingNum
+	}
+
+	skip := (page - 1) * limit
+	max := page * limit
+
+	var issues []*models.Issue
+	for i, depMeta := range deps {
+		if i < skip || i >= max {
+			continue
+		}
+		depMeta.Issue.Repo = &depMeta.Repository
+		issues = append(issues, &depMeta.Issue)
+	}
+
```

Comment:
Usually when you're writing paging code you're attempting to reduce the load on the server and the DB by not getting everything out of the db at once. Here you're collecting almost all of the information then simply throwing it away. Is there not a way of just getting the right dependencies out?

I'd also argue that it seems like the `issue` is only used to get the blocking dependencies before being thrown away - in which case it seems like it's not really needed so perhaps we could just get the blocking issues out directly. In particular `GetIssueWithAttrsByIndex` is loading all of the comments and attachments of the base issue to simply then throw them out! BlockedByDependencies does not need the attributes to be loaded so you don't need that.

In some ways these APIs are unlikely to be called much so the fact that they're not very efficient is not really relevant but it's a point that we should be considering.